"""Rules to run GATK Haplotype Caller

"""

__author__ = "Josh Waterfall <waterfalljj@mail.nih.gov>"
__license__ = "MIT"

GATK_VERSION="3.3-0"

rule gatk_haplotypecaller:
    input: 
        bam = expand("mapping/{{reference}}/samples/{sample}.md.realigned.recal.bam",sample=config['samples'].keys()),
        bai = expand("mapping/{{reference}}/samples/{sample}.md.realigned.recal.bam.bai",sample=config['samples'].keys()),
        dbsnp = config['resources']['dbsnp'],
        reference = lambda wc: config['references'][wc.reference]
    output:
        "variants/{reference}/bwa/gatk_hc/variants.vcf"
    params:
        batch = "-l nodes=1:gpfs -q ccr"
    threads:
        8
    version: GATK_VERSION
    run: 
    	 bams = " ".join('-I ' + bam for bam in input.bam)
    	 shell("module load GATK/{version} && java -Xmx4g -jar $GATKJAR -T HaplotypeCaller -nct {threads} {bams} -R {input.reference} --dbsnp {input.dbsnp} -o {output}")
